### ISSUE TITLE
`react/no-unused-prop-types` false positive when defining type of destructured props [TS]

### ISSUE THREAD

**Issue by Winggo** [yannickcr/eslint-plugin-react#2689]
example1:
```typescript
const profileState = useSelector(
  ({ profile }:
   { profile: ProfileState }) => // 'profile' PropType is defined but prop is never used eslint(react/no-unused-prop-types)
     profile,
);
```

example2:
```typescript
const handleVerifySubmit = ({
  otp,
}: {
  otp: string, // 'otp' PropType is defined but prop is never used eslint(react/no-unused-prop-types)
}) => {
  dispatch(
    verifyOTPPhone({
      otp,
    }),
  );
};
```
I get false positives when defining the type of a destructured prop. Would greatly appreciate any help!

eslint@6.8.0
eslint-plugin-react@7.19.0
typescript@3.7.2

**Comment by MaciekBaron** [comment]
This doesn't seem to be resolved.

    export const RecipeList = ({ data } : { data: any }) => (
      <ul>
        { data.allWordpressPost.edges.map(({ title, slug }: { title: string, slug: string}) => (
          <li>
            {title}
            {' - '}
            {slug}
          </li>
        ))}
      </ul>
    );

I still get `'title' PropType is defined but prop is never used`.

**Comment by ljharb** [comment]
@MaciekBaron are you on v7.20.3? if so, could you file a new issue?

**Comment by MaciekBaron** [comment]
Thanks! I thought I did but now when I did a clean install the issue seems resolved 👍
